pkg/rancher-desktop/main/update/MSIUpdater.ts

```diff
@@ -131,25 +131,27 @@ export default class MsiUpdater extends NsisUpdater {
    * shouldElevate indicates whether we need elevation to install the update.
    */
   protected get shouldElevate(): boolean {
-    // Unfortunately, we seem to be able to write to the application directory
-    // even when installed elevated (possibly due to VirtualStore).  So we rely
-    // on the existence of the privileged service executable to determine
-    // whether we were installed elevated (since we drop the executed when
-    // non-elevated to avoid also installing the privileged service).
-    const checkPath = path.join(paths.resources, 'win32', 'internal', 'privileged-service.exe');
-
-    this._logger.debug?.(`Checking if elevation is needed via ${ checkPath }...`);
+    let key: any = null;
+    let isAdmin: boolean = false;
+
     try {
-      // Unfortunately, doInstall is synchronous, so we need to use the sync
-      // version of access functions.
-      fs.accessSync(checkPath, fs.constants.F_OK);
-      this._logger.debug?.('Elevation is required.');
+      key = reg.openKey(reg.HKLM, 'SOFTWARE', reg.Access.READ);
+
+      if (key) {
+        const parsedValue = reg.getValue(key, 'SUSE\\RancherDesktop', 'AdminInstall');
 
-      return true;
-    } catch (ex) {
-      this._logger.debug?.(`Elevation is not required: ${ ex }`);
+        isAdmin = parsedValue !== null;
 
-      return false;
+        return isAdmin;
+      } else {
+        this._logger.debug?.(`Failed to open registry key: HKEY_LOCAL_MACHINE\SOFTWARE: ${ key }/${ isAdmin }`);
+      }
+    } catch (error) {
+      console.error(`Error accessing registry: ${ error }`);
```

Comment:
Eh, this should be `this._logger` too, but I might be too lazy to make you change it…